### PR TITLE
PERF: improve perf. of Categorical.searchsorted

### PR DESCRIPTION
**Pull request by topper-123**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Improves performance of ``Categorical.searchsorted`` by avoiding expensive data convertions.

```python
>>> n = 100_000
>>> c = pd.Categorical(['a'] * n + ['b'] * n + ['c'] * n)
>>> %timeit c.searchsorted('b')
259 µs ± 2.95 µs per loop  # master
5.5 µs ± 165 ns per loop  # this PR
>>> %timeit c.searchsorted(['b', 'c'])
240 µs ± 4.24 µs per loop  # master
9.9 µs ± 166 ns per loop  # this PR
```

Also, ``CategoricalIndex.searchsorted`` now calls ``self.values.searchsorted`` directly instead of going through ``algorithms.searchsorted``, which always ends up calling ``self.values.searchsorted`` anyway. This ends up getting performance to 5.5 µs instead of 12 µs.

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -162,6 +162,7 @@ Performance improvements
 - Performance improvement in :meth:`DataFrame.corr` when ``method`` is ``"spearman"`` (:issue:`28139`)
 - Performance improvement in :meth:`DataFrame.replace` when provided a list of values to replace (:issue:`28099`)
 - Performance improvement in :meth:`DataFrame.select_dtypes` by using vectorization instead of iterating over a loop (:issue:`28317`)
+- Performance improvement in :meth:`Categorical.searchsorted` and  :meth:`CategoricalIndex.searchsorted` when searching for a single scalar value (:issue:`XXXXX`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v1.0.0.rst]: Just reference the PR as the issue

Comment (topper-123) [reply doc/source/whatsnew/v1.0.0.rst]: Yeah, fixed.

File: pandas/core/arrays/categorical.py [hunk 2]
```diff
@@ -1399,14 +1399,12 @@ def memory_usage(self, deep=False):
...
     def searchsorted(self, value, side="left", sorter=None):
-        from pandas.core.series import Series
-
-        codes = _get_codes_for_values(Series(value).values, self.categories)
-        if -1 in codes:
-            raise KeyError("Value(s) to be inserted must be in categories.")
-
-        codes = codes[0] if is_scalar(value) else codes
-
+        if is_scalar(value):
```
Comment (jreback) [inline pandas/core/arrays/categorical.py]: lgtm, i would add a comment here that this is perf sensitive

**Review by jreback — APPROVED** [review]
lgtm, small comment, ping on green.

**Comment by topper-123** [comment]
Comments addressed.

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**